ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt

```diff
@@ -65,9 +67,11 @@ constructor(
     check(loiId.isNotEmpty()) { "work was queued for an empty location of interest ID" }
     Timber.d("Starting media upload for LOI: $loiId")
 
+    val user = userRepository.getAuthenticatedUser()
     val mutations =
       mutationRepository.getSubmissionMutations(
```

Comment:
We don't need to filter by user since only mutations for the current user should be in the db. Instead, we can continue to fetch all and `Timber.w` and ignore those that don't belong to the current user.

Reply:
gotcha. since we're still processing everything in a list at this point, and not one by one, unless I'm missing something, I think the only way to do this is with a filter, it's just a matter of where we want to apply said filter.

I've moved it into the worker instead.